pkg/models/devops/devops.go

```diff
@@ -137,13 +137,15 @@ func NewDevopsOperator(client devops.Interface,
 }
 
 func convertToHttpParameters(req *http.Request) *devops.HttpParameters {
+	// clone request to prevent concurrent header write
```

Comment:
There is no write action in this function.

Reply:
Please see the following codes:

https://github.com/kubesphere-sigs/ks-devops/blob/5a7d050f52217be2251c1ffe377f0adb76c0c3c1/pkg/client/devops/jenkins/pure_request.go#L52-L59

If two threads share the same header, the race bug would occur and go tool would complain a fatal error.

Reply:
I know that. But there is no write action in this function. How do you explain it? So the problem does not come from this function.